src/app/core/guards/auth-guard/auth.guard.ts

```diff
@@ -1,20 +1,21 @@
 import { Injectable } from "@angular/core";
 import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot } from "@angular/router";
-import { Observable } from "rxjs";
-import { tap } from "rxjs/operators";
+import { Observable, of } from "rxjs";
+import { switchMap, take } from "rxjs/operators";
 
 import { AuthService } from "../../services/index";
+import { StoreFacadeService } from "../../store-facade/store-facade.service";
 
 @Injectable()
 export class AuthGuard implements CanActivate {
     private authService: AuthService;
     private router: Router;
-    private isAuthenticated: Observable<boolean>;
+    private storeFacadeService: StoreFacadeService;

```

Comment:
storeFacade is a pattern. You should use mode specific name, for example, AuthStoreFacade

Reply:
well, I used on storeFacade for two services. I understand that it is risking to become "god-like" service but for the sake of saving time i will avoid refactoring this